File: common/internal_model2/src/main/scala/weco/catalogue/internal_model/locations/AccessMethod.scala
package weco.catalogue.internal_model.locations

import enumeratum.{Enum, EnumEntry}

sealed trait AccessMethod extends EnumEntry { this: AccessMethod =>
  def name: String = this.getClass.getSimpleName.stripSuffix("$")
}

object AccessMethod extends Enum[License] {
  val values = findValues
  assert(
    values.size == values.map { _.id }.toSet.size,
    "IDs for AccessMethod are not unique!"
  )

  case object OpenShelves extends AccessMethod

  case object ViewOnline extends AccessMethod

  case object OnlineRequest extends AccessMethod
  case object ManualRequest extends AccessMethod

  case object NotRequestable extends AccessMethod
}

File: catalogue_graph/src/models/pipeline/access_method.py
from typing import Literal

from models.pipeline.serialisable import SerialisableModel

AccessMethodValue = Literal[
    "OnlineRequest", "ManualRequest", "NotRequestable", "ViewOnline", "OpenShelves"
]


class AccessMethod(SerialisableModel):
    type: AccessMethodValue


OnlineRequest = AccessMethod(type="OnlineRequest")
ManualRequest = AccessMethod(type="ManualRequest")
NotRequestable = AccessMethod(type="NotRequestable")
ViewOnline = AccessMethod(type="ViewOnline")
OpenShelves = AccessMethod(type="OpenShelves")
